symbol_exporter/ast_package_symbol_extractor.py

```diff
@@ -0,0 +1,66 @@
+import ast
+from pathlib import Path
+from typing import Union
+
+from symbol_exporter.ast_symbol_extractor import SymbolFinder
+
+
+def is_package(directory: Path) -> bool:
+    # Only handles regular packages, not namespace packages. See PEP-420.
```

Comment:
To some extent we do handle PEP-420, since it will just be farther down in the dir structure. We will need to handle one off python files though

Reply:
Maybe, currently we only look through the same dir structure, we may miss namespace packages not in the same dir structure.

Specifically section [Differences between namespace packages and regular packages](https://www.python.org/dev/peps/pep-0420/#differences-between-namespace-packages-and-regular-packages)
> Portions of namespace packages need not all come from the same directory structure, or even from the same loader.
> Regular packages are self-contained: all parts live in the same directory hierarchy.

Also
> Namespace packages have no `__init__.py` module

We only walk a directory if it contains a `__init__.py` module in it.
I think we should add an issue and worry about this later.